simpledotorg/simple-server
# DHIS2 aggregate data export (proof of concept)

**Story card:** [ch3369](https://app.clubhouse.io/simpledotorg/story/3369/use-api-to-bulk-insert-aggregate-data)

## Because

We need to test using the DHIS2 API to bulk insert aggregate data for each facility, each month

## This addresses

This uses the `dhis2` Ruby gem that is relatively unmaintained but still works. It sends 9 basic indicators:

- Cumulative registrations
- Monthly registrations
- Cumulative assigned patients
- Cumulative assigned patients (adjusted, excludes <3 month registrations)
- Controlled patients
- Uncontrolled patients
- Missed visits
- Dead patients
- LTFU patients

## Test instructions

1. Copy the sandbox DB locally
2. run `rails dhis2:export`
3. Confirm the data was sent by querying the Pivot Table app: http://dhis2-sandbox.simple.org/dhis-web-pivot/
